### PR TITLE
Initial working role

### PR DESCRIPTION
**Pull request by fujexo**

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -1 +1,33 @@
 ---
+
+# The database icinga2 will export it's configuration to.
```
Comment (karras) [inline defaults/main.yml]: s/it's/its/

File: defaults/main.yml [hunk 2]
```diff
@@ -1 +1,33 @@
...
+# default is "icinga2"
+icinga2_web_database_name: "icinga2"
+
+# The database user icinga2 will export it's configuration to.
```
Comment (karras) [inline defaults/main.yml]: s/it's/its/

File: defaults/main.yml [hunk 3]
```diff
@@ -1 +1,33 @@
...
+# default is "icinga2"
+icinga2_web_database_user: "icinga2"
+
+# The database password icinga2 will export it's configuration to.
```
Comment (karras) [inline defaults/main.yml]: s/it's/its/

File: defaults/main.yml [hunk 4]
```diff
@@ -1 +1,33 @@
...
+# default is "password"
+icinga2_web_database_pass: "password"
+
+# The database host icinga2 will export it's configuration to.
```
Comment (karras) [inline defaults/main.yml]: s/it's/its/

File: defaults/main.yml [hunk 5]
```diff
@@ -1 +1,33 @@
...
+# default is "localhost"
+icinga2_web_database_host: "127.0.0.1"
+
+# The database port icinga2 will export it's configuration to.
```
Comment (karras) [inline defaults/main.yml]: s/it's/its/

File: defaults/main.yml [hunk 6]
```diff
@@ -1 +1,33 @@
...
+# default is "localhost"
```
Comment (karras) [inline defaults/main.yml]: Copypasta error? What's the actual default port?

File: defaults/main.yml [hunk 7]
```diff
@@ -1 +1,33 @@
...
+icinga2_web_database_port: "3306"
+
+icinga2_master_database_name: "icinga2"
```
Comment (karras) [inline defaults/main.yml]: Please add var descriptions here as well.

File: defaults/main.yml [hunk 8]
```diff
@@ -1 +1,33 @@
 ---
+
+# The database icinga2 will export it's configuration to.
+# default is "icinga2"
+icinga2_web_database_name: "icinga2"
```
Comment (karras) [inline defaults/main.yml]: Remove the double ticks unless they're actually required to parse it into a string (e.g. when working with booleans) or YML syntax requires it.

File: handlers/main.yml [hunk 9]
```diff
@@ -1 +1,6 @@
 ---
+
+- name: restart php-fpm
```
Comment (karras) [inline handlers/main.yml]: Adjust the handler name as defined in our standard (https://docs.adfinis-sygroup.ch/public/ansible-guide/styling_guide.html).

File: handlers/main.yml [hunk 10]
```diff
@@ -1 +1,6 @@
...
+  service:
+    name: rh-php71-php-fpm.service
```
Comment (karras) [inline handlers/main.yml]: is `.service` required here?

File: meta/main.yml [hunk 11]
```diff
@@ -1,24 +1,18 @@
...
-        - wheezy
-        - jessie
-        - stretch
-    - name: Ubuntu
-      versions:
-        - trusty
-        - xenial
     - name: EL
       versions:
         - 6
```
Comment (karras) [inline meta/main.yml]: I guess it should only support EL7 right?

File: tasks/configuration.yml [hunk 12]
```diff
@@ -1 +1,134 @@
 ---
+
+# - name:
```
Comment (karras) [inline tasks/configuration.yml]: Remove this leftover...

File: tasks/configuration.yml [hunk 13]
```diff
@@ -1 +1,134 @@
...
+- name: configure php timezone
+  lineinfile:
+    path: /etc/opt/rh/rh-php71/php.ini
+    regexp: '^date.timezone( )?='
+    line: 'date.timezone = Europe/Zurich'
```
Comment (karras) [inline tasks/configuration.yml]: Can you put the timezone into a defaults variable?

File: tasks/configuration.yml [hunk 14]
```diff
@@ -1 +1,134 @@
...
+  notify: restart php-fpm
+
+- name: enable icinga2 ido feature
+  template:
+    src: templates/etc/icinga2/features-enabled/ido-mysql.conf.j2
+    dest: /etc/icinga2/features-enabled/ido-mysql.conf
+    owner: icinga
+    group: icinga
+    mode: 0640
```
Comment (karras) [inline tasks/configuration.yml]: Please define the SELinux context and label

File: tasks/configuration.yml [hunk 15]
```diff
@@ -1 +1,134 @@
...
+
+- name: Set httpd_can_network_connect_db flag on and keep it persistent across reboots
+  seboolean:
+    name: httpd_can_network_connect_db
+    state: yes
+    persistent: yes
+
+- name: check if icingaweb2 table in mysql exists
```
Comment (karras) [inline tasks/configuration.yml]: Is this idempotent?

File: tasks/configuration.yml [hunk 16]
```diff
@@ -1 +1,134 @@
...
+  shell: 'mysql {{icinga2_web_database_name }} -e "SHOW TABLES;" | grep "icingaweb_user"'
+  register: icinga_master_imported
```
Comment (karras) [inline tasks/configuration.yml]: Please stick to our naming convention defined in the standard for registered variables: https://docs.adfinis-sygroup.ch/public/ansible-guide/styling_guide.html#

File: tasks/configuration.yml [hunk 17]
```diff
@@ -1 +1,134 @@
...
+
+- name: import icinga2 ido database schema
+  mysql_db:
+    name: '{{ icinga2_master_database_name }}'
+    login_host: '{{ icinga2_master_database_host }}'
+    login_port: '{{ icinga2_master_database_port }}'
+    login_user: '{{ icinga2_master_database_user }}'
+    login_password: '{{ icinga2_master_database_pass }}'
+    state: import
+    target: '/usr/share/icinga2-ido-mysql/schema/mysql.sql'
```
Comment (karras) [inline tasks/configuration.yml]: Only use ticket when required

File: tasks/configuration.yml [hunk 18]
```diff
@@ -1 +1,134 @@
...
+
+
+# configure icingaweb2
+- name: configure icingaweb2 basics
+  template:
+    src: templates/etc/icingaweb2/config.ini.j2
+    dest: /etc/icingaweb2/config.ini
+    owner: root
+    group: icingaweb2
+    mode: 0660
```
Comment (karras) [inline tasks/configuration.yml]: Please define the SELinux context and label

File: tasks/configuration.yml [hunk 19]
```diff
@@ -1 +1,134 @@
...
+
+- name: configure icingaweb2 resources
+  template:
+    src: templates/etc/icingaweb2/resources.ini.j2
+    dest: /etc/icingaweb2/resources.ini
+    owner: root
+    group: icingaweb2
+    mode: 0660
```
Comment (karras) [inline tasks/configuration.yml]: Please define the SELinux context and label

File: tasks/configuration.yml [hunk 20]
```diff
@@ -1 +1,134 @@
...
+
+- name: configure icingaweb2 roles
+  template:
+    src: templates/etc/icingaweb2/roles.ini.j2
+    dest: /etc/icingaweb2/roles.ini
+    owner: root
+    group: icingaweb2
+    mode: 0660
```
Comment (karras) [inline tasks/configuration.yml]: Please define the SELinux context and label

File: tasks/configuration.yml [hunk 21]
```diff
@@ -1 +1,134 @@
...
+
+- name: configure icingaweb2 authentication
+  template:
+    src: templates/etc/icingaweb2/authentication.ini.j2
+    dest: /etc/icingaweb2/authentication.ini
+    owner: root
+    group: icingaweb2
+    mode: 0660
```
Comment (karras) [inline tasks/configuration.yml]: Please define the SELinux context and label

File: tasks/configuration.yml [hunk 22]
```diff
@@ -1 +1,134 @@
...
+
+
+# icinga2 api setup
+# /etc/icinga2/conf.d/api-users.conf
```
Comment (karras) [inline tasks/configuration.yml]: Remove those leftover comments

File: tasks/configuration.yml [hunk 23]
```diff
@@ -1 +1,134 @@
...
+
+# enable and configure the icingaweb2 monitoring module
+- name: check if icingaweb2 table in mysql exists
+  shell: 'mysql {{icinga2_web_database_name }} -e "SHOW TABLES;" | grep "icingaweb_user"'
```
Comment (karras) [inline tasks/configuration.yml]: Is this idempotent?

File: tasks/configuration.yml [hunk 24]
```diff
@@ -1 +1,134 @@
...
+
+- name: import icingaweb2 database schema
+  mysql_db:
+    name: '{{ icinga2_web_database_name }}'
+    login_host: '{{ icinga2_web_database_host }}'
+    login_port: '{{ icinga2_web_database_port }}'
+    login_user: '{{ icinga2_web_database_user }}'
+    login_password: '{{ icinga2_web_database_pass }}'
+    state: import
+    target: '/usr/share/doc/icingaweb2/schema/mysql.schema.sql'
```
Comment (karras) [inline tasks/configuration.yml]: Only use ticks when required...

File: defaults/main.yml [hunk 25]
```diff
@@ -1 +1,33 @@
...
+
+# The database port icinga2 will export it's configuration to.
+# default is "localhost"
+icinga2_web_database_port: "3306"
+
+icinga2_master_database_name: "icinga2"
+icinga2_master_database_host: "127.0.0.1"
+icinga2_master_database_port: "3306"
+icinga2_master_database_user: "icinga2"
+icinga2_master_database_pass: "password"
```
Comment (karras) [inline defaults/main.yml]: Always prefix all vars with the role name...

File: tasks/installation.yml [hunk 26]
```diff
@@ -1 +1,14 @@
...
+    name: '{{ icinga2_web_scl }}'
+    state: present
+  when:
+    ansible_distribution == 'CentOS'
+
+- name: install icinga2 web packages
+  package:
+    name: '{{ icinga2_web_packages }}'
+    state: present
+  when: ansible_distribution == 'CentOS'
```
Comment (karras) [inline tasks/installation.yml]: Remove that condition, it works on any distribution if the correct vars are set ;)

File: vars/RedHat.yml [hunk 27]
```diff
@@ -1 +1,12 @@
 ---
+
+# The package list needed for icinga web2
+
+icinga2_web_scl:
```
Comment (karras) [inline vars/RedHat.yml]: Add var descriptions for every variable

File: README.rst [hunk 28]
```diff
@@ -1,15 +1,15 @@
...
-  :target: https://travis-ci.org/adfinis-sygroup/ansible-role-_template
+.. image:: https://img.shields.io/travis/adfinis-sygroup/ansible-role-icinga2_web.svg?style=flat-square
+  :target: https://travis-ci.org/adfinis-sygroup/ansible-role-icinga2_web
 
-.. image:: https://img.shields.io/badge/galaxy-adfinis--sygroup._template-660198.svg?style=flat-square
-  :target: https://galaxy.ansible.com/adfinis-sygroup/_template
+.. image:: https://img.shields.io/badge/galaxy-adfinis--sygroup.icinga2_web-660198.svg?style=flat-square
+  :target: https://galaxy.ansible.com/adfinis-sygroup/icinga2_web
 
 A brief description of the role goes here.
```
Comment (hairmare) [inline README.rst]: > A brief description of the role goes here.

:sob:

File: tasks/configuration.yml (hunk 23 above)
Comment (fujexo) [reply tasks/configuration.yml]: Hopefully. We can discuss this at the office, maybe you see another way of checking if the import already happened

File: defaults/main.yml (hunk 25 above)
Comment (fujexo) [reply defaults/main.yml]: I'll probably move this to the icinga2_master role, this was pushed by accident. Thanks for pointing that out!

File: tasks/installation.yml [hunk 29]
```diff
@@ -1 +1,70 @@
...
+    name: '{{ icinga2_web_icingaweb2_database_name }}'
+    login_host: '{{ icinga2_web_icingaweb2_database_host }}'
+    login_port: '{{ icinga2_web_icingaweb2_database_port }}'
+    login_user: '{{ icinga2_web_icingaweb2_database_user }}'
+    login_password: '{{ icinga2_web_icingaweb2_database_pass }}'
+    state: import
+    target: '/usr/share/doc/icingaweb2/schema/mysql.schema.sql'
+  when: icinga2_web_register_icingaweb2_imported.rc == 1
+
+- name: ensure icingaweb2 modules directory is present
```
[the remark this answers is not in the document]
Comment (fujexo) [reply tasks/installation.yml]: I'm not sure how to achieve this as it seems to be hardcoded..
https://github.com/Icinga/icingaweb2/blob/184d4afdbdb945fd0ce25c499ea368999a8b8105/library/Icinga/Application/ApplicationBootstrap.php#L411

File: tasks/configuration.yml [hunk 30]
```diff
@@ -1 +1,88 @@
...
+    serole: object_r
+    setype: icingaweb2_config_t
+    selevel: s0
+
+- name: create icingaweb2 admin password hash
+  shell: openssl passwd -1 '{{ icinga2_web_admin_pass }}'
+  register: icinga2_web_register_admin_hash
+  when: icinga2_web_register_icingaweb2_imported.rc == 1
+
+- name: insert icingaweb2 admin password into database
```
[the remark this answers is not in the document]
Comment (fujexo) [reply tasks/configuration.yml]: What do you think if I rename the task name to `create first icingaweb2 admin user`

[the remark this answers is not in the document]
Comment (MrStrix) [reply tasks/configuration.yml]: you're right, all ok :)

File: README.rst [hunk 31]
```diff
@@ -35,34 +37,30 @@ should be mentioned here as well.
 Dependencies
```
Comment (karras) [inline README.rst]: Please describe the role vars.

File: meta/main.yml (hunk 11 above)
Comment (karras) [reply meta/main.yml]: Status?

File: tasks/configuration.yml (hunk 14 above)
Comment (karras) [reply tasks/configuration.yml]: Status?

File: tasks/installation.yml [hunk 32]
```diff
@@ -1 +1,72 @@
 ---
+
+- name: install icinga2 web packages
+  package:
+    name: '{{ icinga2_web_packages }}'
+    state: present
+  register: icinga2_web_register_packages_installed
+  until: icinga2_web_register_packages_installed is succeeded
```
Comment (karras) [inline tasks/installation.yml]: Can you add comment why `until` is required?

File: tasks/installation.yml [hunk 33]
```diff
@@ -1 +1,72 @@
...
+    target: '/usr/share/doc/icingaweb2/schema/mysql.schema.sql'
+  when: icinga2_web_register_icingaweb2_imported.rc == 1
+
+- name: ensure icingaweb2 modules directory is present
+  file:
+    name: /etc/icingaweb2/enabledModules
+    state: directory
+    owner: root
+    group: icingaweb2
+    mode: 0755
```
Comment (karras) [inline tasks/installation.yml]: SELinux?

File: templates/etc/icinga2/features-available/ido-mysql.conf.j2 [hunk 34]
```diff
@@ -0,0 +1,13 @@
+// {{ ansible_managed }}
```
Comment (karras) [inline templates/etc/icinga2/features-available/ido-mysql.conf.j2]: Add a comment regarding the icingaweb2 package version this config is based on?

File: templates/etc/icingaweb2/authentication.ini.j2 [hunk 35]
```diff
@@ -0,0 +1,5 @@
+;{{ ansible_managed }}
```
Comment (karras) [inline templates/etc/icingaweb2/authentication.ini.j2]: Add a comment regarding the icingaweb2 package version this config is based on?

File: templates/etc/icingaweb2/config.ini.j2 [hunk 36]
```diff
@@ -0,0 +1,13 @@
+;{{ ansible_managed }}
```
Comment (karras) [inline templates/etc/icingaweb2/config.ini.j2]: Add a comment regarding the icingaweb2 package version this config is based on?

File: templates/etc/icingaweb2/groups.ini.j2 [hunk 37]
```diff
@@ -0,0 +1,5 @@
+;{{ ansible_managed }}
```
Comment (karras) [inline templates/etc/icingaweb2/groups.ini.j2]: Add a comment regarding the icingaweb2 package version this config is based on?

File: templates/etc/icingaweb2/modules/monitoring/backends.ini.j2 [hunk 38]
```diff
@@ -0,0 +1,5 @@
+;{{ ansible_managed }}
```
Comment (karras) [inline templates/etc/icingaweb2/modules/monitoring/backends.ini.j2]: Add a comment regarding the icingaweb2 package version this config is based on?

File: tasks/installation.yml (hunk 32 above)
Comment (fujexo) [reply tasks/installation.yml]:
```
[fujexo@hippaforalkus ansible-role-icinga2_web]$ ansible-lint -R -r ../lint-rules .
[405] Remote package tasks should have a retry
/home/fujexo/git/adfinis/ansible/ansible-role-icinga2_web/tasks/installation.yml:3
Task/Handler: install icinga2 web packages
```

Ansible lint complains about this. Should I add a comment to the task?

Comment (fujexo) [reply tasks/installation.yml]: the other solution is to disable this specific lint error and remove the `until`

Comment (eyenx) [reply tasks/installation.yml]: How can this be done?

Comment (fujexo) [reply tasks/installation.yml]: Adding an `.ansible-lint` configuration file like this:
```yaml
skip_list:
  - '405'
```

Comment (fujexo) [reply tasks/installation.yml]: Discussed with @karras. we will skip this lint number

Comment (eyenx) [reply tasks/installation.yml]: thx

**Pull request merged**